src/mixins/labels.js

```diff
@@ -0,0 +1,115 @@
+            const spriteMaterial = new THREE.SpriteMaterial({ map: texture });
+            const sprite = new THREE.Sprite(spriteMaterial);
+            sprite.name = name;
+            sprite.scale.set(0.25, 0.25, 0.25);
+            sprite.visible = this.areLabelsShown;
+            return sprite;
+        }
+
+        adjustLabelsToCameraPosition() {
+            const { scene, camera } = this;
+            const cellCenter = this.getCellCenter();
+            const atomGroups =
+                scene
+                    .getObjectByName(ATOM_GROUP_NAME)
+                    ?.parent.children.filter((object) => object.name.includes(ATOM_GROUP_NAME)) ||
+                [];
+            atomGroups.forEach((group) => {
+                const labels = group.children.filter((child) => child instanceof THREE.Sprite);
+                labels.forEach((label) => {
+                    const atomUUID = label.name.replace(/label-for-/, "");
+                    const atom = scene.getObjectByProperty("uuid", atomUUID);
+                    const atomRadius = atom.geometry.parameters.radius;
+
+                    const atomPosition = atom.position.clone().add(group.position);
+                    const vectorToCamera = this.isCameraOrthographic
+                        ? new THREE.Vector3(
+                              camera.position.x - cellCenter[0],
+                              camera.position.y - cellCenter[1],
+                              camera.position.z - cellCenter[2],
+                          )
+                        : new THREE.Vector3(
+                              camera.position.x - atomPosition.x,
+                              camera.position.y - atomPosition.y,
+                              camera.position.z - atomPosition.z,
+                          );
+
+                    const clampedVectorToCamera = vectorToCamera
+                        .clone()
+                        .clampLength(atomRadius, atomRadius);
+                    const newLabelPosition = atom.position.clone().add(clampedVectorToCamera);
+                    label.position.copy(newLabelPosition);
+                });
+            });
+        }
+
+        toggleLabels() {
+            this.areLabelsShown = !this.areLabelsShown;
+
+            const { scene } = this;
+            const atomGroups =
+                scene
+                    .getObjectByName(ATOM_GROUP_NAME)
+                    ?.parent.children.filter((object) => object.name.includes(ATOM_GROUP_NAME)) ||
+                [];
+            const labels = atomGroups
+                .map((group) => group.children.filter((child) => child instanceof THREE.Sprite))
+                .flat();
+            labels.forEach((label) => {
+                label.visible = this.areLabelsShown;
+            });
+
+            this.render();
+        }
+
+        createLabel(message, name) {
+            return this.createSprite(message, name);
```

Comment:
Not sure we need this function, we could just use `createLabelSprite` directly per the above

Reply:
Yes, this is definitely an echo of earlier functionality regrouping.